### PR TITLE
GH-43143: [C++][Parquet] Default initialize some parquet metadata variables

### PR DESCRIPTION
**Pull request by mapleFU**
### Rationale for this change

Default initialize some parquet metadata variables

### What changes are included in this PR?

Default initialize some parquet metadata variables

### Are these changes tested?

Covered by existing

### Are there any user-facing changes?

no

* GitHub Issue: #43143

### CODE REVIEW

File: cpp/src/parquet/schema.h [hunk 1]
```diff
@@ -464,7 +465,7 @@ class PARQUET_EXPORT SchemaDescriptor {
   // Root Node
   schema::NodePtr schema_;
   // Root Node
-  const schema::GroupNode* group_node_;
+  const schema::GroupNode* group_node_{nullptr};
```
Comment (wgtmac) [inline cpp/src/parquet/schema.h:468]: This seems unnecessary.

File: cpp/src/parquet/schema.h [hunk 2]
```diff
@@ -416,8 +417,8 @@ class PARQUET_EXPORT ColumnDescriptor {
 // TODO(wesm): this object can be recomputed from a Schema
 class PARQUET_EXPORT SchemaDescriptor {
  public:
-  SchemaDescriptor() {}
-  ~SchemaDescriptor() {}
+  SchemaDescriptor() = default;
+  ~SchemaDescriptor() = default;
```
Comment (raulcd) [inline cpp/src/parquet/schema.h:421]: for a C++ newbie, what is the difference here?

Comment (mapleFU) [reply cpp/src/parquet/schema.h:421]: no differences here

File: cpp/src/parquet/schema.h (hunk 1 above)
Comment (mapleFU) [reply cpp/src/parquet/schema.h]: I change this to a comment

**Review by pitrou — APPROVED** [review]
+1, LGTM

**Review by kou — APPROVED** [review]
+1

**Pull request merged**